### ISSUE TITLE
Nested grid-containers cause excess padding

### ISSUE THREAD

**Issue by JarvisH** [zurb/foundation-sites#10624]
Paddings were added to grid-containers recently ([https://github.com/zurb/foundation-sites/pull/10371](https://github.com/zurb/foundation-sites/pull/10371)), but in certain contexts where grid-containers may be nested inside other grid-containers, excess paddings are present.

This could be compared to the old system of rows inside rows, which had a negative offset left/right to compensate.

Solution:

`.grid-container .grid-container { padding:0 }`

#### How to reproduce this bug:

Create a grid-container inside another.

**Comment by mattbag** [comment]
Why would you nest grid-container?

**Comment by JarvisH** [comment]
If working in the context of a CMS, where elements use grid-containers to ensure a max-width problems arise when placing an element inside another grid column. Dynamically using grid-containers in such a context would be tricky.

**Comment by mattbag** [comment]
Hey I still don't get it, if you have an outer grid-container, are you saying you want anyway to do:
```
<div class="grid-container">
    <div class="grid-x">
        <div class="cell">
  <!-- component nested ? -->
            <div class="grid-container">
                <div class="grid-x">
                    <div class="cell">
                        [component]
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
```
If I got it right I would just go with a page-container on the outside, give it everything beside the padding...

**Comment by JarvisH** [comment]
Hi.

Your code example above is correctly showing what I meant. A main page wrapper would work, but that then prevents you from easily adding elements that should be expanded to the full available window width, e.g. an image-teaser or similar.

All in all it's no big deal, I managed to add the one line of css from my initial post, which solves the issue, I just found it to be counterintuitive when comparing to the old system of row classes, where this issue was not present :-)

**Comment by mattbag** [comment]
Yes, I used the wrong word, I didn't mean an actual page container, but a limited width container.
Still I don't see the need of nesting grid-container, especially based on what you just said, you use the class for rows that need to be limited, and none if you want full width for images video etx.
And that is the reason why it's different from previous version where everything was on row, now it's more modular.

**Comment by JarvisH** [comment]
Absolutely, it would make sense just to leave out the grid-containers when they are not needed, but this brings us back to the original problem, that when using predefined templates (i.e. a CMS) either we would need to programmatically figure out when to leave out said grid-containers or the CMS-editors would need to manually change a setting/template depending on where fx. a text-element is placed, so my point is really more about convenience than an actual bug.

**Comment by JarvisH** [comment]
Another thing I just realized is that grid-padding-x nested inside a grid-margin-x will have wrong alignments, when not wrapping grid-padding-x inside a grid-container, which again will only align properly if nested grid-containers have zero padding.

Codepen example here: [Codepen](https://codepen.io/jarvisH/pen/oGgaRR)

Or are grid-padding-x and grid-margin-x containers not meant to be mixed?

**Comment by mattbag** [comment]
Yes it makes way more sense to me to have the option full or not, since we are talking about a site theme apparently. If you take wp themes for instance.

For the second, I think if you want to use them together, it means you want double sizing. So far I haven't had need of margins for columns at all...

**Comment by Anke** [comment]
I have to agree with JarvisH. Every so often there are scenarios requiring the use of nested grids and then the margins and paddings on the outer left and outer right colums are a problem. 

Or if using xy grid only for parts of the content, e.g. when using a full-width headline and/or introduction text succeeded by an article set in columns. In this case the headline and introduction would be wider than the article set in columns. Unless I'd set headline and introduction in a one-column-grid, which is counter-productive if you want as little code as possible.  

Sometimes I'm helping myself by repealing a margin (or padding) of grid-margin-x:first-child/last-child, but all in all I think the paddings/margins of outer columns are a pain and should have been gotten more consideration in the development. - Or am I missing something?
